tests/unit/test_messages_emitter.py

```diff
@@ -109,15 +109,41 @@ def test_init_quietish(mode, tmp_path, monkeypatch):
     assert handler.mode == mode
 
 
+def test_init_verbose_mode(tmp_path, monkeypatch):
+    """Init the class in verbose mode."""
+    # avoid using a real log file
+    fake_logpath = str(tmp_path / FAKE_LOG_NAME)
+    monkeypatch.setattr(messages, "_get_log_filepath", lambda appname: fake_logpath)
+
+    greeting = "greeting"
+    emitter = Emitter()
+    with patch("craft_cli.messages._Printer") as mock_printer:
+        emitter.init(EmitterMode.VERBOSE, "testappname", greeting)
+
+    assert emitter._mode == EmitterMode.VERBOSE
+    log_locat = f"Logging execution to {fake_logpath!r}"
+    assert mock_printer.mock_calls == [
+        call(fake_logpath),  # the _Printer instantiation, passing the log filepath
+        call().show(None, "greeting"),  # the greeting, only sent to the log
+        call().show(sys.stderr, greeting, use_timestamp=False, end_line=True, avoid_logging=True),
+        call().show(sys.stderr, log_locat, use_timestamp=False, end_line=True, avoid_logging=True),
+    ]
+
+    # log handler is properly setup
+    logger = logging.getLogger("")
+    (handler,) = [x for x in logger.handlers if isinstance(x, _Handler)]
+    assert handler.mode == EmitterMode.VERBOSE
+
+
 @pytest.mark.parametrize(
     "mode",
     [
-        EmitterMode.VERBOSE,
+        EmitterMode.DEBUG,
         EmitterMode.TRACE,
     ],
 )
-def test_init_verboseish(mode, tmp_path, monkeypatch):
-    """Init the class in some verbose-ish mode."""
+def test_init_developer_modes(mode, tmp_path, monkeypatch):
```

Comment:
I like your phrases, "developer modes" and "quiet modes".  At first I thought I missed something in the source code or documentation, but couldn't find any references.

It took me a few minutes to understand the meaning.  Sometimes quiet modes = "quiet and brief", but other times quiet modes = "everything except trace".  I think a brief comment or description in the test suite about those two phrases would be nice.